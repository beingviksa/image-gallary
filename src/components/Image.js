import { useState } from "react";
import Modal from "./Modal";

const Image = ({ path, alt, shape, description }) => {
  const [showModal, setShowModal] = useState(false);
  const handleClick = () => {
    setShowModal(true);
  };
  return (
    <div className="image-box">
      <img
        data-testid={`${alt}-image`}
        src={path}
        alt={alt}
        className={`image ${shape}`}
        onClick={handleClick}
      />
      {showModal && <Modal setShowModal={setShowModal}>{description}</Modal>}
    </div>
  );
};

export default Image;
