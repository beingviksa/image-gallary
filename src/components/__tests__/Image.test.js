import { fireEvent, render } from "@testing-library/react";
import Image from "../Image";

describe("<Image/>", () => {
  const props = {
    path: "src/images/tajmahal.jpeg",
    alt: "TajMahal",
    shape: "circle",
    description:
      "The Taj Mahal, is an Islamic ivory-white marble mausoleum on the right bank of the river Yamuna in the Indian city of Agra.",
  };

  it("render the Image Component", () => {
    const { container } = render(<Image {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("show modal on click of Image", () => {
    const { getByTestId } = render(<Image {...props} />);
    const placeImage = getByTestId("TajMahal-image");

    fireEvent.click(placeImage);

    const modal = getByTestId("modal-container");
    expect(modal).toBeInTheDocument();
  });
});
