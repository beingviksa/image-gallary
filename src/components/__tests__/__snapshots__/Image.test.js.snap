// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> render the Image Component 1`] = `
<div>
  <div
    class="image-box"
  >
    <img
      alt="TajMahal"
      class="image circle"
      data-testid="TajMahal-image"
      src="src/images/tajmahal.jpeg"
    />
  </div>
</div>
`;
